modules/syncMinimongo.js

```diff
@@ -3,59 +3,122 @@
-            //     webContents.send('minimongo-removed', id);
-            // else
-            if(!webContents)
-                ipc.send('minimongo-removed', id);
+            coll.insert(args.fields);
         }
     });
 
-    // receive
-    ipc.on('minimongo-add', function(event, args) {
+    ipc.on('minimongo-changed', function(event, args) {
+        log.trace('minimongo-changed', args._id);
+
+        var _id = args._id;
+
+        var item = coll.findOne({_id: _id});
 
-        if(!db.findOne(args.id)) {
-            // log.info('add', args.id);
-            args.fields._id = args.id;
-            db.insert(args.fields);
+        if (item) {
+            for (var k in args.fields) {
+                item[k] = args.fields[k];
+            }
+
+            coll.update(item);
+        } else {
+            log.error('Item not found in db', _id);
         }
     });
-    ipc.on('minimongo-changed', function(event, args) {
 
-        // log.info('updated', args.id, args.fields);
-        db.update(args.id, {$set: args.fields});
+    ipc.on('minimongo-removed', function(event, args) {
+        log.trace('minimongo-removed', args._id);
+
+        var _id = args._id;
+
+        var item = coll.findOne({_id: _id});
+
+        if (item) {
+            coll.remove(item);
+        } else {
+            log.error('Item not found in db', _id);
+        }
     });
-    ipc.on('minimongo-removed', function(event, id) {
 
-        db.remove(id.toString());
+    // get all data (synchronous)
+    ipc.on('minimongo-reloadSync', function(event) {
+        var docs = coll.find();
+
+        log.debug('minimongo-reloadSync, no. of docs:', docs.length);
+
+        docs = docs.map(function(doc) {
+            var ret = {};
+
+            for (var k in doc) {
+                if ('meta' !== k && '$loki' !== k) {
+                    ret[k] = doc[k];
+                }
+            }
+
+            return ret;
+        });
+
+        event.returnValue = JSON.stringify(docs);
     });
-};
\ No newline at end of file
+};
+
+
+
+exports.frontendSync = function(coll) {
+    var ipc = electron.ipcRenderer;
+
+    var repopulating = true;
+
+    coll.find().observeChanges({
+        'added': function(id, fields){            
+            if (!repopulating) {
+                ipc.send('minimongo-add', {_id: id, fields: fields});
+            }
+        },
+        'changed': function(id, fields){
+            if (!repopulating) {
+                ipc.send('minimongo-changed', {_id: id, fields: fields});
+            }
+        },
+        removed: function(id) {
+            if (!repopulating) {
+                ipc.send('minimongo-removed', {_id: id});
+            }
+        }
+    });
+
+    let dataStr = ipc.sendSync('minimongo-reloadSync');
+
+    if (dataStr) {
+        console.debug('Repopulate collection with backend data: ', coll._name);
+
+        coll.remove({});
+
+        JSON.parse(dataStr).forEach(function(record) {
```

Comment:
Maybe we wrap this in a
```js
Tracker.afterFlush(function(){
   JSON.parse(dataStr).forEach(function(record) {...
})
```
`afterFlush`, means it will wait for all reactive actions to finish and execute this in the next reactive cycle. This is a bit like `setTimeout(fund, 0)`, but in meteors reactive cycles

Otherwise, im fine with this PR, we can fix the issue later, if we don't fix it with this.